frontends/verilog/verilog_parser.y

```diff
@@ -1933,10 +1951,32 @@ typedef_decl:
 			rewriteAsMemoryNode(astbuf1, $5);
 		}
 		addTypedefNode($4, astbuf1); }
-	| TOK_TYPEDEF non_wire_data_type type_name ';'   { addTypedefNode($3, $2); }
+	| TOK_TYPEDEF enum_struct_type type_name ';'   { addTypedefNode($3, $2); }
 	;
 
-non_wire_data_type:
+data_type:
+	hierarchical_type_id {
+		$$ = new AstNode(AST_WIRE);
+		$$->is_logic = true;
+		$$->is_custom_type = true;
+		$$->children.push_back(new AstNode(AST_WIRETYPE));
+		$$->children.back()->str = *$1;
+		delete $1;
+	} |
+	integer_vector_type opt_signedness_default_unsigned {
+		$$ = new AstNode(AST_WIRE);
+		$$->is_logic = true;
```

Comment:
Should this use `is_logic` for `TOK_LOGIC` and `is_reg` for `TOK_REG`?

Reply:
TL;DR I am deliberately trying to make `reg` be the same as `logic` here. I think this is most clearly stated in the 1800-2017 standard in section 6.11.2 third paragraph when it says "**logic** and **reg** denote the same type." 

Specifically I expect `reg` and `var reg` to have have the same behavior as `logic` and `var logic` in SV mode where both continuous and procedural assignment are allowed but not both. In Verilog 2005 a continuous assignment to reg is not allowed. Since we are using typedef I assume we must be in SV mode.

Here is are some examples using verilator, iverilog, and yosys in Verilog 2005 and SystemVerilog mode to help me demonstrate. 

```systemverilog
module test;
    reg r;
    assign r = 0;
endmodule
```

This is illegal in Verilog 2005:

```sh
verilator --lint-only +1364-2005ext+v test.v
%Error-CONTASSREG: test.v:3:12: Continuous assignment to reg, perhaps intended wire (IEEE 1364-2005 6.1; Verilog only, legal in SV): 'r'
                                : ... In instance test1
    3 |     assign r = 0;
      |            ^
%Error: Exiting due to 1 error(s)
        ... See the manual and https://verilator.org for more assistance.
```

```sh
iverilog -g2005 -tnull test.v
test.v:3: error: reg r; cannot be driven by primitives or continuous assignment.
1 error(s) during elaboration.
```

```sh
yosys -q test.v
Warning: reg '\r' is assigned in a continuous assignment at test.v:3.12-3.17.
```

But should be legal in SystemVerilog:

```sh
verilator --lint-only +1800-2017ext+sv test.v
```
```sh
iverilog -tnull -g2012 test.v
```

```sh
yosys -q -p "read_verilog -sv test.v"
Warning: reg '\r' is assigned in a continuous assignment at test.v:3.12-3.17.
```

Should be legal with typedef too:

```systemverilog
module test;
    typedef reg R;
    R r;
    assign r = 0;
endmodule
```

```sh
verilator --lint-only +1800-2017ext+sv test.sv
```
```sh
iverilog -tnull -g2012 test.sv
```
```sh
yosys -q test.sv
```

Like a mentioned in one of the commit messages I think the only effect is the warning message. In the future, maybe `is_logic` and `is_reg` should be combined and only warn if there is continuous and procedural assignment.

Reply:
You are absolutely right that `reg` should behave like `logic`. I agree we should consider disabling the `reg` continuous assignment warning in SystemVerilog mode, but I think we should keep it to a separate (and more complete) change. With this change as is, the following two snippets behave differently:
```SystemVerilog
typedef reg T;
T x;
assign x = 1;
```
```SystemVerilog
reg x;
assign x = 1;
```
Further, even once we disable the warning in SV mode, I can imagine someone might quite reasonably want a flag to turn it back on to retain the old behavior.

For these reasons, I think the parser should retain the `reg` and `logic` distinction within typedefs, at least for now.

Reply:
OK no problem. I will update tonight or tomorrow.